logrotate.c

```diff
@@ -545,8 +546,11 @@ static int runScript(const struct logInfo *log, const char *logfn, const char *l
         exit(1);
     }
 
-    wait(&rc);
-    return rc;
+    if (wait(&rc) > 0 && WIFEXITED(rc)) {
```

Comment:
Could you please take this opportunity and replace with `wait(3)` with `waitpid(3)`.
```c
waitpid(pid, &rc, 0);
```